src/Middleware/OutputCaching/src/IOutputCachePolicyProvider.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.OutputCaching;
+
+/// <summary>
+/// A provider to get output cache policies for a given request.
```

Comment:
Doesn't it feel weird to say "for a given request" while the  provider is registered as singleton? The notion of request only appears in the NamedPolicy (OutputCacheContext).

Related question, will this be invoked for each request or only once for the application, in which case maybe registering it as transient would make sense (use and forget).